lib/certification/index.js

```diff
@@ -0,0 +1,505 @@
+  const providerInfo = {};
+
+  try {
+    Object.assign(providerInfo, JSON.parse(fs.readFileSync(pathToConfigFile)));
+  } catch (err) {
+    handleError(`Could not read provider info! pathToConfigFile: ${pathToConfigFile}`);
+  }
+
+  const { providerUoi, configs } = providerInfo;
+
+  if (!(providerUoi && providerUoi?.length)) handleError('providerUoi is required!');
+  if (!(configs && configs?.length)) handleError('configs must contain valid configurations');
+
+  for await (const config of configs) {
+    const { providerUsi, recipientUoi } = config;
+
+    const RECIPIENT_PATH = join(
+      process.cwd(),
+      buildRecipientEndorsementPath({
+        providerUoi,
+        providerUsi,
+        recipientUoi,
+        endorsementName,
+        version
+      })
+    );
+
+    const pathToConfigFile = join(RECIPIENT_PATH, CONFIG_FILE_NAME);
+
+    if (!(providerUsi && providerUsi?.length)) handleError(`providerUsi is missing from the given config: ${config}!`);
+    if (!(recipientUoi && recipientUoi?.length)) handleError(`recipientUoi is missing from the given config: ${config}!`);
+
+    try {
+      //archive existing results if they already exist
+      archiveEndorsement({ providerUoi, providerUsi, recipientUoi, endorsementName, version });
+
+      //create recipient path for new results
+      fs.mkdirSync(RECIPIENT_PATH, { recursive: true });
+
+      //build test config and write it to the appropriate path
+      const testingConfig = buildTestingConfig(config);
+      if (!testingConfig && Object.keys(testingConfig)?.length) {
+        handleError(`There was a problem creating a RESOScript config for recipientUoi: ${recipientUoi}`);
+      }
+      fs.writeFileSync(pathToConfigFile, testingConfig);
+
+      // clean up testing files before each job is run
+      cleanUpCertificationFiles();
+
+      // if the task fails, it will throw an error
+      console.log('Running Data Dictionary metadata tests...');
+      await executeCommanderMetadataTest({ pathToWebApiCommander: WEB_API_COMMANDER_PATH, version, pathToConfigFile, fromCli, strictMode });
+
+      if (runAllTests) {
+        const replicationStateServiceInstance = createReplicationStateServiceInstance();
+
+        let pathToMetadataReportJson;
+        try {
+          pathToMetadataReportJson = await processDataDictionaryMetadataFiles(RECIPIENT_PATH);
+        } catch (err) {
+          handleError(err, { terminate: false });
+        }
+
+        try {
+          await executeDataDictionaryTestSteps({
+            config,
+            version,
+            pathToMetadataReportJson,
+            replicationStateServiceInstance,
+            fromCli,
+            strictMode
+          });
+        } catch (err) {
+          handleError(err, { terminate: false });
+        }
+      }
+    } catch (err) {
+      if (fromCli) {
+        const commanderLogPath = join(getWebApiCommanderPath(), COMMANDER_LOG_FILE_NAME);
+        const { size } = fs.statSync(commanderLogPath);
```

Comment:
This can be an unhandled exception inside a catch if `commanderLogPath` doesn't exist.